### PR TITLE
Use astropy nightly wheel in devdeps CI tests

### PR DESCRIPTION
**Pull request by larrybradley**
### Description

Closes https://github.com/spacetelescope/jdaviz/issues/1457

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -38,7 +40,7 @@ description =
 # The following provides some specific pinnings for key packages
 deps =
     # NOTE: Add/remove as needed
-    devdeps: git+https://github.com/astropy/astropy.git#egg=astropy
+    devdeps: astropy>=0.0.dev0
```
[the remark this answers is not in the document]
Comment (pllim) [reply tox.ini:43]: Also, the numpy dev command has to be consistent if we switch to this syntax. And I was thinking about scipy dev too.

[the remark this answers is not in the document]
Comment (larrybradley) [reply tox.ini:43]: You could specify the whole pip command in the "commands" section.  I copied what sunpy uses. :)

[the remark this answers is not in the document]
Comment (pllim) [reply tox.ini:43]: I am not quite a fan of the way sunpy do it, as it is a little hard to read. Is there a reason why their way is superior?

[the remark this answers is not in the document]
Comment (larrybradley) [reply tox.ini]: I changed to use the pip command.

File: tox.ini [hunk 2]
```diff
@@ -59,6 +58,7 @@ extras =
 
 commands =
     devdeps: pip install -U --pre --only-binary :all: -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
+    devdeps: pip install -U -i https://pypi.anaconda.org/astropy/simple astropy --pre
```
Comment (pllim) [inline tox.ini:61]: Might as well? Or do you think it is not a good idea?

```suggestion
    devdeps: pip install -U --pre --only-binary :all: -i https://pypi.anaconda.org/scipy-wheels-nightly/simple scipy
    devdeps: pip install -U -i https://pypi.anaconda.org/astropy/simple astropy --pre
```

File: tox.ini (hunk 1 above)
[the remark this answers is not in the document]
Comment (pllim) [reply tox.ini]: Thanks!

File: tox.ini (hunk 2 above)
Comment (larrybradley) [reply tox.ini:62]: why not?

**Comment by pllim** [comment]
Remote data timeout is unrelated.

**Pull request merged**